src/_pytest/cacheprovider.py

```diff
@@ -5,39 +5,41 @@
 ignores the external pytest-cache
 """
 from __future__ import absolute_import, division, print_function
-
 from collections import OrderedDict
 
 import py
 import six
+import attr
 
 import pytest
 import json
-import os
 from os.path import sep as _sep, altsep as _altsep
+import shutil
+
+from . import paths
+from .compat import _PY2 as PY2
 
 
+@attr.s
 class Cache(object):
 
-    def __init__(self, config):
-        self.config = config
```

Comment:
i realized that removing config and trace is not a good change as its "public" api